### PR TITLE
run tests with MySQL, PostgreSQL, SQLite

### PR DESCRIPTION
**Pull request by craue**
Different approach compared to #420. This makes it possible to run specific tests (only those which actually use a database) against several database platforms. I definitely like that the mechanism is easier to understand and tests themselves are kept clean.

What I don't like is that a failed step causes following steps to be skipped so if tests fail with MySQL they won't be run with PostgreSQL and SQLite. Also the `SYMFONY_DEPRECATIONS_HELPER` will be harder to use with 4 PHPUnit steps.

I wonder why `PhotoUploadFlowTest` fails with PostgreSQL. 🤔 This at least proves that testing against several database platforms is a good idea. 😂

### CODE REVIEW

**Comment by craue** [comment]
PostgreSQL is fine now, at least with PHP > 7.3. I'm gonna remove support for PHP 7.3 but not within this PR.

Awaiting feedback from @franmomu. 😊

**Comment by craue** [comment]
Alright, now I also got all PHPUnit steps running even in case of failures. 🥳

File: Tests/Storage/DoctrineStorageTest.php [hunk 1]
```diff
@@ -5,18 +5,30 @@
...
 
+	/**
+	 * @var Connection
+	 */
+	private $conn;
+
+	protected function getConnection() : Connection {
+		return $this->conn;
+	}
+
```
Comment (franmomu) [inline Tests/Storage/DoctrineStorageTest.php:31]: Looks like this is not used anymore

**Review by franmomu — APPROVED** [review]
LGTM!

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -91,7 +118,25 @@ jobs:
...
+        if: always()
+        run: vendor/bin/phpunit -v --coverage-clover build/logs/clover-postgresql.xml --group run-with-multiple-databases,run-with-multiple-databases-only
+        env:
+          DB_FLAVOR: postgresql
+
+      - name: run PHPUnit with SQLite
+        if: always()
+        run: vendor/bin/phpunit -v --coverage-clover build/logs/clover-sqlite.xml --group run-with-multiple-databases,run-with-multiple-databases-only
+        env:
+          DB_FLAVOR: sqlite
```
[the remark this answers is not in the document]
Comment (craue) [reply .github/workflows/tests.yml:139]: But then I'd have to decide with which PHP/Symfony versions they'd be run with. As of now, cross coverage is a lot better.

**Pull request merged**